File: Backend/src/main/java/Backend/service/ResourceService.java
package Backend.service;

import Backend.exception.ResourceNotFoundException;
import Backend.model.ResourceModel;
import Backend.repository.ResourceRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class ResourceService {

    @Autowired
    private ResourceRepository resourceRepository;

    public List<ResourceModel> getAllResources() {
        return resourceRepository.findAll();
    }

    public ResourceModel getResourceById(String id) {
        return resourceRepository.findById(id)
                .orElseThrow(() ->
                        new ResourceNotFoundException("Resource not found: " + id));
    }

    public List<ResourceModel> getResourcesByType(String type) {
        return resourceRepository.findByType(type);
    }

    public List<ResourceModel> getResourcesByStatus(String status) {
        return resourceRepository.findByStatus(status);
    }

    public List<ResourceModel> getResourcesByLocation(String location) {
        return resourceRepository.findByLocation(location);
    }

    public ResourceModel addResource(ResourceModel resource) {
        return resourceRepository.save(resource);
    }

    public ResourceModel updateResource(String id, ResourceModel updated) {
        ResourceModel existing = resourceRepository.findById(id)
                .orElseThrow(() ->
                        new ResourceNotFoundException("Resource not found: " + id));

        existing.setName(updated.getName());
        existing.setType(updated.getType());
        existing.setLocation(updated.getLocation());
        existing.setCapacity(updated.getCapacity());
        existing.setStatus(updated.getStatus());
        existing.setAvailableFrom(updated.getAvailableFrom());
        existing.setAvailableTo(updated.getAvailableTo());
        existing.setDescription(updated.getDescription());

        return resourceRepository.save(existing);
    }

    public void deleteResource(String id) {
        ResourceModel existing = resourceRepository.findById(id)
                .orElseThrow(() ->
                        new ResourceNotFoundException("Resource not found: " + id));
        resourceRepository.delete(existing);
    }
}

File: frontend/src/services/resourceService.js
import axios from 'axios';

const api = axios.create({
    baseURL: 'http://localhost:8085/api',
    withCredentials: true,
    headers: { 'Content-Type': 'application/json' }
});
const API_URL = "http://localhost:8085/api/resources";

// GET ALL
export const getAllResources = async () => {
    const response = await api.get('/resources');
    return response.data;
}

// GET BY ID
export const getResourceById = async (id) => {
    const response = await api.get(`/resources/${id}`);
    return response.data;
}

// GET BY TYPE
export const getResourcesByType = async (type) => {
    const response = await api.get(`/resources/type/${type}`);
    return response.data;
}

// GET BY STATUS
export const getResourcesByStatus = async (status) => {
    const response = await api.get(`/resources/status/${status}`);
    return response.data;
}

// GET BY LOCATION
export const getResourcesByLocation = async (location) => {
    const response = await api.get(`/resources/location/${location}`);
    return response.data;
}

// POST - ADD
export const addResource = async (resource) => {
    const response = await api.post('/resources', resource);
    return response.data;
}

// PUT - UPDATE
export const updateResource = async (id, resource) => {
    const response = await api.put(`/resources/${id}`, resource);
    return response.data;
}

// DELETE
export const deleteResource = async (id) => {
    await api.delete(`/resources/${id}`);
}